zcash_client_sqlite/src/for_async/init.rs

```diff
@@ -0,0 +1,182 @@
+    wdb: &WalletDbAsync<P>,
+) -> Result<(), rusqlite::Error>
+where
+    P: Clone + Send + Sync,
+{
+    let wdb = wdb.inner.clone();
+    async_blocking(move || {
+        let wdb = wdb.lock().unwrap();
+
+        wdb.conn.execute(
+            "CREATE TABLE IF NOT EXISTS accounts (
+            account INTEGER PRIMARY KEY,
+            extfvk TEXT NOT NULL,
+            address TEXT NOT NULL
+        )",
+            [],
+        )?;
+        wdb.conn.execute(
+            "CREATE TABLE IF NOT EXISTS blocks (
+            height INTEGER PRIMARY KEY,
+            hash BLOB NOT NULL,
+            time INTEGER NOT NULL,
+            sapling_tree BLOB NOT NULL
+        )",
+            [],
+        )?;
+        wdb.conn.execute(
+            "CREATE TABLE IF NOT EXISTS transactions (
+            id_tx INTEGER PRIMARY KEY,
+            txid BLOB NOT NULL UNIQUE,
+            created TEXT,
+            block INTEGER,
+            tx_index INTEGER,
+            expiry_height INTEGER,
+            raw BLOB,
+            FOREIGN KEY (block) REFERENCES blocks(height)
+        )",
+            [],
+        )?;
+        wdb.conn.execute(
+            "CREATE TABLE IF NOT EXISTS received_notes (
+            id_note INTEGER PRIMARY KEY,
+            tx INTEGER NOT NULL,
+            output_index INTEGER NOT NULL,
+            account INTEGER NOT NULL,
+            diversifier BLOB NOT NULL,
+            value INTEGER NOT NULL,
+            rcm BLOB NOT NULL,
+            nf BLOB NOT NULL UNIQUE,
+            is_change INTEGER NOT NULL,
+            memo BLOB,
+            spent INTEGER,
+            FOREIGN KEY (tx) REFERENCES transactions(id_tx),
+            FOREIGN KEY (account) REFERENCES accounts(account),
+            FOREIGN KEY (spent) REFERENCES transactions(id_tx),
+            CONSTRAINT tx_output UNIQUE (tx, output_index)
+        )",
+            [],
+        )?;
+        wdb.conn.execute(
+            "CREATE TABLE IF NOT EXISTS sapling_witnesses (
+            id_witness INTEGER PRIMARY KEY,
+            note INTEGER NOT NULL,
+            block INTEGER NOT NULL,
+            witness BLOB NOT NULL,
+            FOREIGN KEY (note) REFERENCES received_notes(id_note),
+            FOREIGN KEY (block) REFERENCES blocks(height),
+            CONSTRAINT witness_height UNIQUE (note, block)
+        )",
+            [],
+        )?;
+        wdb.conn.execute(
+            "CREATE TABLE IF NOT EXISTS sent_notes (
+            id_note INTEGER PRIMARY KEY,
+            tx INTEGER NOT NULL,
+            output_index INTEGER NOT NULL,
+            from_account INTEGER NOT NULL,
+            address TEXT NOT NULL,
+            value INTEGER NOT NULL,
+            memo BLOB,
+            FOREIGN KEY (tx) REFERENCES transactions(id_tx),
+            FOREIGN KEY (from_account) REFERENCES accounts(account),
+            CONSTRAINT tx_output UNIQUE (tx, output_index)
+        )",
+            [],
+        )?;
+        Ok(())
```

Comment:
This can be replaced by `crate::wallet::init::init_wallet_db(&wdb)`, this way, if we update `librustzcash` and the original function changes we get the changes here too.

Reply:
done